### ISSUE TITLE
How to divide the data used by the Hyperopt into a train set and a validation set

### ISSUE THREAD

**Issue by wilmerwang** [freqtrade/freqtrade#8866]
## Describe your environment

  * Operating system:  Ubuntu 22
  * Python Version: Python 3.10
  * CCXT version: _____ (`pip freeze | grep ccxt`)
  * Freqtrade Version: ____ (`freqtrade -V` or `docker compose run --rm freqtrade -V` for Freqtrade running in docker)
  
## Your question

When I use `Hyperopt` to Optimize  strategies,  it often train and test on the same data, which are often overfit and lead to poor performance on other test data. Is it possible to add a data splitting feature to divide the dataset into training and validation sets, so that we can use the model with best-performing on the validation set to improve the model's generalization ability?

**Comment by xmatthias** [comment]
You can specify a timerange for this (e.g. `--timerange 20220101-20230101`) - and then use a different timerange for the subsequent backtest on the improved model (e.g. `--timerange 20230101-`).

**Comment by wilmerwang** [comment]
> You can specify a timerange for this (e.g. `--timerange 20220101-20230101`) - and then use a different timerange for the subsequent backtest on the improved model (e.g. `--timerange 20230101-`).

Thanks for your reply, that is a good idea. As I optimize the program, I will obtain a series of models with continuously decreasing training loss during 20220101-20230101. How can I determine which model results in the lowest validation loss during 20230101-now? Is there a feature to calculate the validation loss at the end of each training epoch, so that we can select the model with the lowest validation loss?

**Comment by xmatthias** [comment]
with "validation loss", i assume you'll mean the hyperopt Loss function.
it's output is really just there to give hyperopt a direction to work towards, NOT to be used as a measure per se.

what you should instead be looking at is the output of the backtest report.
you can get this for each hyperopt result via `freqtrade hyperopt-show -n <epochid>` (this will also write the parameters file - so a subsequent backtest will use THESE parameters).
you can get that for the test period by simply running backtesting against that period with the "best" training parameters
